evm_scilla_js_tests/test/BlockProperties.ts

```diff
@@ -0,0 +1,65 @@
+import {expect} from "chai";
+import hre from "hardhat";
+import {Contract} from "ethers";
+import {ethers} from "hardhat";
+
+describe("Block Properties", function () {
+  let contract: Contract;
+  before(async function () {
+    contract = await hre.deployContract("BlockAndTransactionProperties");
+  });
+
+  it("Should be deployed successfully @block-1", async function () {
+    expect(contract.address).to.be.properAddress;
+  });
+
+  it("should return the correct block number", async function () {
+    const contractBlockNumber = await contract.getBlockNumber();
+    const currentBlockNumber = await ethers.provider.getBlockNumber();
+
+    expect(contractBlockNumber).to.equal(currentBlockNumber);
+  });
+
+  it("should return the correct block timestamp", async function () {
+    const contractBlockTimestamp = await contract.getTimestamp();
+    const currentBlock = await ethers.provider.getBlock("latest");
+
+    expect(contractBlockTimestamp).to.equal(currentBlock.timestamp);
+  });
+
+  it("should return the correct base fee", async function () {
+    const contractBaseFee = await contract.getBaseFee();
+
+    expect(contractBaseFee).to.equal(await hre.ethers.provider.getGasPrice());
+  });
+
+  it("should return the correct chain id", async function () {
+    const contractChainId = await contract.getChainId();
+    const network = await ethers.provider.getNetwork();
+
+    expect(contractChainId).to.equal(network.chainId);
+  });
+
+  it("should return the correct coinbase", async function () {
+    const contractCoinbase = await contract.getCoinbase();
+
+    expect(contractCoinbase).to.equal("0x0000000000000000000000000000000000000000");
+  });
+
+  it("should return the correct gas limit", async function () {
+    const contractGasLimit = await contract.getGasLimit();
+    const currentBlock = await ethers.provider.getBlock("latest");
+
+    expect(contractGasLimit).to.equal(currentBlock.gasLimit);
+  });
+
+  it("should emit blockhash correctly", async function () {
```

Comment:
Can we add test cases for `blockhash` with the current block number and with a block number more than 256 blocks in the past? I believe both of these should return zero. Maybe a future block number test too for good measure?